### PR TITLE
 🦋 Allow overwrite of non-POS, non-bank transfer payment on POS sessi…

### PR DESCRIPTION
**Pull request by ithiame**
…on #1570

### CODE REVIEW

File: .vscode/settings.json
Comment (coyotte508) [inline .vscode/settings.json:1]: Revert???

Comment (ithiame) [reply .vscode/settings.json:1]: already deleted ??

File: src/lib/server/orders.ts [hunk 1]
```diff
@@ -989,7 +990,9 @@ export async function createOrder(
 
 		let orderPayment: OrderPayment | undefined = undefined;
 		if (paymentMethod) {
-			const expiresAt = paymentMethodExpiration(paymentMethod);
+			const expiresAt = params.paymentTimeOut
+				? paymentMethodExpiration(paymentMethod, { paymentTimeout: params.paymentTimeOut })
+				: paymentMethodExpiration(paymentMethod);
```
Comment (coyotte508) [inline src/lib/server/orders.ts:995]:
```suggestion
			const expiresAt = paymentMethodExpiration(paymentMethod, { paymentTimeout: params.paymentTimeOut });
```

File: src/routes/(app)/checkout/+page.svelte [hunk 2]
```diff
@@ -496,6 +497,30 @@
...
+								name="paymentTimeOut"
+								class="form-input"
+								min="1"
+								value={data.desiredPaymentTimeout}
+								disabled={!changePaymentTimeOut}
+								required={changePaymentTimeOut}
+							/>
+						</div>
+					</label>
+				{/if}
```
Comment (coyotte508) [inline src/routes/(app)/checkout/+page.svelte:523]: This should only be available to POS users, currently everyone can see

File: src/routes/(app)/checkout/+page.server.ts [hunk 3]
```diff
@@ -373,7 +374,13 @@ export const actions = {
 				delete billingInfo.billing.vatNumber;
 			}
 		}
-
+		const desiredPayment = z
+			.object({
+				paymentTimeOut: z.number({ coerce: true }).optional()
```
Comment (coyotte508) [inline src/routes/(app)/checkout/+page.server.ts:379]: Make it `min(1)` and positive integer

Comment (ithiame) [reply src/routes/(app)/checkout/+page.server.ts]: min(1) gives error if the param is not changed

File: src/lib/server/orders.ts [hunk 4]
```diff
@@ -1300,14 +1303,13 @@ async function generatePaypalPaymentInfo(params: {
...
 }
 
-function paymentMethodExpiration(paymentMethod: PaymentMethod) {
+function paymentMethodExpiration(paymentMethod: PaymentMethod, opts?: { paymentTimeout?: number }) {
 	return paymentMethod === 'point-of-sale' || paymentMethod === 'bank-transfer'
 		? undefined
-		: paymentMethod === 'lightning' &&
-		  isPhoenixdConfigured() &&
+		: (paymentMethod === 'lightning' && isPhoenixdConfigured() && opts?.paymentTimeout) ||
 		  runtimeConfig.desiredPaymentTimeout > 60
```
Comment (coyotte508) [inline src/lib/server/orders.ts:1310]:
```suggestion
		: paymentMethod === 'lightning' && isPhoenixdConfigured() && (opts?.paymentTimeout ??
		  runtimeConfig.desiredPaymentTimeout) > 60
```

File: src/routes/(app)/checkout/+page.server.ts [hunk 5]
```diff
@@ -373,7 +374,13 @@ export const actions = {
 				delete billingInfo.billing.vatNumber;
 			}
 		}
-
+		const desiredPayment = z
+			.object({
+				paymentTimeOut: z.number({ coerce: true }).int().optional()
```
Comment (coyotte508) [inline src/routes/(app)/checkout/+page.server.ts:379]:
```suggestion
				paymentTimeOut: z.number({ coerce: true }).int().min(1).optional()
```

**Review by coyotte508 — APPROVED** [review]

**Pull request merged**